chainer/functions/loss/discriminative_loss.py

```diff
@@ -0,0 +1,439 @@
+from chainer.backends import cuda
+from chainer.functions.array.broadcast import broadcast_to
+from chainer.functions.array.cast import cast
+from chainer.functions.array.concat import concat
+from chainer.functions.array.expand_dims import expand_dims
+from chainer.functions.array.reshape import reshape
+from chainer.functions.array.stack import stack
+from chainer.functions.array.transpose import transpose
+from chainer.functions.math.average import average
+from chainer.functions.math.basic_math import absolute
+from chainer.functions.math.maximum import maximum
+from chainer.functions.math.sum import sum as c_sum
+
+import warnings
+
+
+class DiscriminativeMarginBasedClusteringLoss(object):
+
+    """Discriminative margin-based clustering loss function
+
+    This is the implementation of the following paper:
+    https://arxiv.org/abs/1802.05591
```

Comment:
Is the citation correct? The paper in #4513 is https://arxiv.org/pdf/1708.02551.pdf

Reply:
Fixed.